### ISSUE TITLE
[ENH]

### ISSUE THREAD

**Issue by arnaujc91** [sktime/sktime#4943]
**Is your feature request related to a problem? Please describe.**
under `forecasting/model_selection/_spilt.py` you have the following code:

```python
if isinstance(y, (pd.DatetimeIndex, pd.PeriodIndex)) and is_int(window_length):
    window_length = y.freq * window_length

``` 
Possibly the frequency of the datetimeIndex can be null but it can be inferred using pandas

**Describe the solution you'd like**
Modify the code to:

```python
if isinstance(y, (pd.DatetimeIndex, pd.PeriodIndex)) and is_int(window_length):
   if y.freq is null:
       try:
           y.freq = y.inferred_freq
      except Exception as e:
           throw Exception time series data frequency could not be inferred.
    window_length = y.freq * window_length
```

Easy to fix

**Comment by fkiraly** [comment]
Would you like to make a pull request? Would be much appreciated!

Optimally, include a test case that fails before the change, but succeeds after the change.

**Comment by arnaujc91** [comment]
On my way to the PR :)

**Comment by fkiraly** [comment]
Is this related to https://github.com/sktime/sktime/pull/5009 ?

**Comment by hazrulakmal** [comment]
Yup. I suppose @arnaujc91 faced the same bug as me and raised this issue with a solution. Could you verify if you are facing the same issue as #4972, which led you to raise this issue? 

If so please do review the PR I raised in #5009 and we can work from there.

**Comment by arnaujc91** [comment]
Hi @hazrulakmal I face exactly the same error. I will take a look at your PR.
